### DISCUSSION TITLE
Put Replicache into offline mode?

### DISCUSSION THREAD

**Discussion by tslocke** [rocicorp/replicache#992] [Q&A] [answered]
Is there any easy way to put replicache into an offline mode, so that it neither pushes or pulls?

Would be useful to isolate front-end issues while debugging.

**Comment by arv** [comment, marked as answer]
- Set `pullURL` and `pushURL` to `''`. These are read/write so clearing them prevents next push/pul.
- We set `pushDelay` to a large number in our tests. This _prevents_ the automatic push after a mutation.
- `pullInterval` is already pretty high (60s) but you can increase this too.
- Other options includes implementing a custom `puller`/`pusher`.

**Comment by tslocke** [comment]
Setting `pullURL` and `pushURL` to `''` didn't work for me - Replicache was still making requests and started getting 404s

Setting a long `pushDelay` and `pullInterval` did work.

**Comment by aboodman** [comment]
You actually need to set them to null, not empty string. Apologies.

**Comment by aboodman** [comment]
Or rather undefined. See:
https://doc.replicache.dev/api/interfaces/ReplicacheOptions#pullurl

**Comment by arv** [comment]
This is confusing me. The code looks like both `undefined` and `''` should work:

```ts
export class Replicache<MD extends MutatorDefs = {}> {
  ...
  private _isPushDisabled() {
    return this.pushURL === '' && this.pusher === defaultPusher;
  }

  private _isPullDisabled() {
    return this.pullURL === '' && this.puller === defaultPuller;
  }
```

@tslocke Do you have a custom `pusher`/`puller`?

**Reply by tslocke** [reply]
Yes I do - just for some logging. I guess that's why none of these options work for me : )

**Reply by arv** [reply]
Interesting. Maybe we can solve the logging needs in some other way?

For now, in your `Puller` you can skip the `response` field of the `PullResponse` and it should be treated as a noop. For your `Pusher` you can return `{httpStatusCode: 200, errorMessage: ''}`
